cmd/thanos/compact.go

```diff
@@ -754,6 +763,9 @@ func (cc *compactConfig) registerFlag(cmd extkingpin.FlagClause) {
 		"as querying long time ranges without non-downsampled data is not efficient and useful e.g it is not possible to render all samples for a human eye anyway").
 		Default("false").BoolVar(&cc.disableDownsampling)
 
+	strategies := strings.Join([]string{string(concurrentDiscovery), string(recursiveDiscovery)}, ", ")
+	cmd.Flag("block-discovery-strategy", "One of "+strategies+". When set to concurrent, stores will concurrently issue one call per directory to discover active blocks in the bucket. The recursive strategy iterates through all objects in the bucket, recursively traversing into each directory. This avoids N+1 calls at the expense of having slower bucket iterations.").
```

Comment:
We revert the current behavior and change back to the old one? Is it worth a changelog?

Reply:
I am no authority on anything, but I would personally like to have a changelog entry on this. Seeing how it was mentioned in different threads how this is a blocker for upgrading Thanos itself for some, I think it is important enough.

Reply:
Good call. I added a changelog entry, let me know if that's sufficient or we should elaborate further.

Reply:
I like it. Good enough for me.